### PR TITLE
Add PROCESS run/read/mock to EUDEMO

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by hsaunders1904 — APPROVED** [review]
Does everything run smoothly in run/read mode? It's all unit tested, but I'm not sure that we've really put this solver through its paces since the refactor.

**Comment by CoronelBuendia** [comment]
> Does everything run smoothly in run/read mode? It's all unit tested, but I'm not sure that we've really put this solver through its paces since the refactor.

Run and mock work fine. OK so run is a little janky in terms of a design point at the moment (irrelevant). read I haven't tried, to be honest. I can add a random IN.DAT and check it?

File: eudemo/eudemo/radial_build.py [hunk 1]
```diff
@@ -23,21 +23,29 @@
 from typing import Dict, TypeVar
 
 from bluemira.base.parameter_frame import ParameterFrame
+from bluemira.codes import process
```
Comment (je-cook) [inline eudemo/eudemo/radial_build.py:26]:
```suggestion
from bluemira.codes import systems_code_solver
```

File: eudemo/eudemo/radial_build.py [hunk 2]
```diff
@@ -23,21 +23,29 @@
...
-    with open(build_config["file_path"], "r") as f:
-        param_values = json.load(f)
-    params.update_values(param_values, source="PROCESS (mock)")
+    Returns
+    -------
+    params: ParameterFrame
+        Updated parameters following the solve.
+    """
+    run_mode = build_config.pop("run_mode", "mock")
+    solver = process.Solver(params, build_config)
```
Comment (je-cook) [inline eudemo/eudemo/radial_build.py:48]:
```suggestion
    solver = systems_code_solver(params, build_config)
```

**Review by je-cook — COMMENTED** [review]
Two minors from me if we're trying to keep it all pluggable

**Comment by CoronelBuendia** [comment]
> > Does everything run smoothly in run/read mode? It's all unit tested, but I'm not sure that we've really put this solver through its paces since the refactor.
> 
> Run and mock work fine. OK so run is a little janky in terms of a design point at the moment (irrelevant). read I haven't tried, to be honest. I can add a random IN.DAT and check it?

OK added MFILE and tested that "read" works too

**Pull request merged**